amazon-msk/cdktf/secure-public-access/main.ts

```diff
@@ -329,14 +363,12 @@ export class ZillaPlusSecurePublicAccessStack extends TerraformStack {
       description: "The public wildcard DNS pattern for bootstrap servers to be used by Kafka clients",
     });
 
-    const publicTlsCertificateKey = new TerraformVariable(this, "public_tls_certificate_key", {
-      type: "string",
-      description: "TLS Certificate Private Key Secret ARN",
-    });
-    // Validate that the Certificate Key exists
-    new DataAwsSecretsmanagerSecretVersion(this, "publicTlsCertificate", {
-      secretId: publicTlsCertificateKey.stringValue,
-    });
+    if (publicTlsCertificateKey.stringValue.startsWith("arn:aws:secretsmanager:")) {
```

Comment:
I think this is not doing what you intended to do. As publicTlsCertificateKey is a TerraformVariable, at synch time it's only a TOKEN. So this conditions only checks the token's value not the actual input. I guess what you want is:
`Fn.startswith(publicTlsCertificateKey.stringValue, "arn:aws:secretsmanager:")`